src/librbd/ImageWatcher.cc

```diff
@@ -844,7 +844,6 @@ void ImageWatcher::handle_payload(const FlattenPayload &payload,
 
 void ImageWatcher::handle_payload(const ResizePayload &payload,
 				  bufferlist *out) {
-  RWLock::RLocker l(m_image_ctx.owner_lock);
```

Comment:
Since the owner lock isn't being held, the exclusive lock can be released.  Before the exclusive lock is released, all pending ops are cleanly canceled.  In this case, w/o the owner lock being held, it's possible for the exclusive lock to be released before the op has a chance to add itself to the pending list.